data/payroll.mk

```diff
@@ -0,0 +1,68 @@
+.INTERMEDIATE : %-with-agencies.csv %-with-data-year.csv \
+	%-with-valid-start-dates.csv %-salary-summed.csv \
+	%-no-salary-omitted.csv %-with-agencies.csv
+
+.PRECIOUS : %-with-agencies.csv data/output/%
+
+define get_record_count
+$$(csvstat --count $(1) | cut -d ' ' -f3)
+endef
+
+.PHONY : import/% amend/%
+
+import/% : data/output/%
+	python manage.py import_data --data_file $< \
+		--reporting_year $$(echo "$<" | grep -Eo "[0-9]{4}")
+
+amend/% : data/output/%
+	python manage.py import_data --data_file $< \
+		--reporting_year $$(echo "$<" | grep -Eo "[0-9]{4}") \
+		--amend
+
+data/output/payroll-actual-%.csv : %-with-data-year.csv
+	# Rename fields.
+	(echo employer,last_name,first_name,title,department,base_salary,extra_pay,date_started,id,_,responding_agency,data_year; \
+	tail -n +2 $<) > $@
+
+%-with-data-year.csv : %-no-salary-omitted.csv
+	# Add required data year field.
+	perl -pe "s/$$/,$$(cut -d '-' -f1 <<< $*)/" $< > $@
+
+%-no-salary-omitted.csv : %-with-valid-start-dates.csv
+	# Remove records where no salary was reported.
+	csvgrep -c base_salary,extra_pay -r '^$$' -i $< > $@
+	N_REMOVED=$$(expr $(call get_record_count,$<) - $(call get_record_count,$@)); \
+	echo "Removed $$N_REMOVED records without salary"
+
+%-with-valid-start-dates.csv : %-with-agencies.csv
+	# Remove invalid dates.
+	cat $< | python data/processors/validate_dates.py > $@
+
+%-with-agencies.csv : %-whitespace-trimmed.csv data/raw/foia-source-lookup.csv
+	# Join standard data with agency lookup and validate that the output file
+	# has the expected number of lines.
+	csvjoin -c id,ID -e IBM852 $^ > $@
+	if [ $(call get_record_count,$<) -eq $(call get_record_count,$@) ]; then \
+		echo "No lines lost in join. Proceeding..."; \
+	else \
+		echo "$< has $$INFILE_LINES lines, but $@ has $$OUTFILE_LINES lines. Generating missing agencies..."; \
```

Comment:
how does $INFILE_LINES and $OUTFILE_LINES get set?

Reply:
@fgregg Whoops, fixed in https://github.com/datamade/bga-payroll/pull/526/commits/10058ca9bbaaf91bcd6455f8ae909277ba5230a1. Need the line continuations in this recipe because the Make runs each command in its own subshell, so they need to be part of the same command as the conditional that references them.

Reply:
word